plaso/parsers/czip_plugins/interface.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+"""czip_interface contains basic interface for compound ZIP plugins within
+Plaso.
+
+CompoundZIPPlugin defines the attributes necessary for registration, discovery
+and operation of plugins for czip files which will be used by CompoundZIPParser.
+"""
+
+from __future__ import unicode_literals
+
+from plaso.lib import errors
+from plaso.parsers import logger
+from plaso.parsers import plugins
+
+class CompoundZIPPlugin(plugins.BasePlugin):
+  """This is and "abstract" class from which plugins should be based."""
+
+  # REQUIRED_ITEMS is a list of items required by a plugin.
+  # This is used to understand whether a plugin is suited for a given compound
+  # ZIP archive.
+  # This is expected to be overridden by actual plugins.
+  # Ex. frozenset(['[Content_Types].xml', '_rels/.rels', 'docProps/core.xml'])
+  REQUIRED_ITEMS = frozenset(['any'])
+
+  NAME = 'czip'
+
+  # pylint: disable=arguments-differ
+  def Process(self, parser_mediator, archive_proxy, **kwargs):
```

Comment:
Archive_proxy seems to add a fair bit of complexity here, with not a lot of utility. It's also not a great name, since this is actually the czip parser. You can just move Read() into the czip plugin interface and not need this.

Reply:
So I just pass `file_object` instead? Or is there a smarter way to access parent parser from within a plugin?

Reply:
Yes, you could parse file_object, or even zip_file. There's no easy way to reference the parent parser from a plugin.